src/sqlfluff/dialects/dialect_clickhouse.py

```diff
@@ -744,63 +743,121 @@ class CreateTableStatementSegment(ansi.CreateTableStatementSegment):
 
     type = "create_table_statement"
 
-    match_grammar: Matchable = Sequence(
-        "CREATE",
-        OneOf(
-            Ref("OrReplaceGrammar"),
-            Ref.keyword("TEMPORARY"),
-            optional=True,
+    match_grammar: Matchable = OneOf(
+        Sequence(
+            "CREATE",
+            Ref("OrReplaceGrammar", optional=True),
+            "TABLE",
```

Comment:
Is it not much simpler to just add `Ref.keyword("TEMPORARY", optional=True)` just before `TABLE`

Reply:
The problem is that this code is valid for TEMPORARY tables
https://fiddle.clickhouse.com/9f82dca6-80d1-439f-b947-eff1c9ac6d98
But it is not valid for default tables
https://fiddle.clickhouse.com/fff310e3-a771-4a4a-b6de-96a85ef150b0

Reply:
Because of this, I made the code separate for CREATE TEMPORARY.
Syntax from the docks
Temporary:
https://clickhouse.com/docs/en/sql-reference/statements/create/table#temporary-tables
Default table:
https://clickhouse.com/docs/en/sql-reference/statements/create/table#syntax-forms